### PR TITLE
CLN: fix all flake8 warnings in pandas/tools

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

File: pandas/tools/pivot.py [hunk 1]
```diff
@@ -24,12 +24,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
...
     data : DataFrame
     values : column to aggregate, optional
-    index : a column, Grouper, array which has the same length as data, or list of them.
-        Keys to group by on the pivot table index.
-        If an array is passed, it is being used as the same manner as column values.
-    columns : a column, Grouper, array which has the same length as data, or list of them.
-        Keys to group by on the pivot table column.
-        If an array is passed, it is being used as the same manner as column values.
+    index : a column, Grouper, array which has the same length as data, or list
+        of them.
```
Comment (jorisvandenbossche) [inline pandas/tools/pivot.py]: this is another numpydoc problem ... 
the type specification should be on one line to have to formatted well in the html docs (as the first line after the colon is formatted in italic), and then second line is the start of the explanation

Comment (jorisvandenbossche) [inline pandas/tools/pivot.py]: In this case, maybe just shorten it a bit? (maybe 'array which has the same length' -> 'array of same length' is enough to keep within line limit?)

But if this happens more, we should maybe look if we can solve this in the numpydoc parser

Comment (wesm) [inline pandas/tools/pivot.py]: Yeah, the original versions contained commingled types and descriptions. Let me do a proper job fixing

**Comment by jorisvandenbossche** [comment]
For the rest, looks good from a quick glance

**Comment by wesm** [comment]
Cool, I'll merge if this build passes

File: pandas/tools/merge.py [hunk 2]
```diff
@@ -245,11 +248,15 @@ def _indicator_post_merge(self, result):
...
 
-        result[self.indicator_name] = Categorical((result['_left_indicator'] + result['_right_indicator']), categories=[1,2,3])
-        result[self.indicator_name] = result[self.indicator_name].cat.rename_categories(['left_only', 'right_only', 'both'])
-
-        result = result.drop(labels=['_left_indicator', '_right_indicator'], axis=1)
+        result[self.indicator_name] = Categorical((result['_left_indicator'] +
+                                                   result['_right_indicator']),
+                                                  categories=[1, 2, 3])
+        result[self.indicator_name] = (
+            result[self.indicator_name]
```
Comment (jreback) [inline pandas/tools/merge.py]: I personally find this formatting very hard to read (though not real sure what's better)

**Comment by jreback** [comment]
lgtm

File: pandas/tools/merge.py (hunk 2 above)
Comment (wesm) [inline pandas/tools/merge.py]: A lot of these formatting issues stem from statements that are too dense. `result[self.indicator_name]` should be a temporary variable here, as should perhaps the list of new categories

**Pull request closed without merging**